File: android/src/main/java/com/getcapacitor/community/classes/options/SendOptions.java
package com.getcapacitor.community.classes.options;

import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;

public class SendOptions {

    @Nullable
    private String message;

    @Nullable
    private String peerID;

    public SendOptions(PluginCall call) {
        @Nullable
        String message = call.getString("message");
        this.setMessage(message);

        @Nullable
        String peerID = call.getString("peerID");
        this.setPeerID(peerID);
    }

    private void setMessage(@Nullable String message) {
        this.message = message;
    }

    private void setPeerID(@Nullable String peerID) {
        this.peerID = peerID;
    }

    @Nullable
    public String getMessage() {
        return message;
    }

    @Nullable
    public String getPeerID() {
        return peerID;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Options/SendOptions.swift
import Foundation
import Capacitor

@objc public class SendOptions: NSObject {
    private var message: String?
    private var peerID: String?

    init(_ call: CAPPluginCall) {
        super.init()

        if let message = call.getString("message") {
            self.setMessage(message)
        }

        if let peerID = call.getString("peerID") {
            self.setPeerID(peerID)
        }
    }

    func setMessage(_ message: String?) {
        self.message = message
    }

    func getMessage() -> String? {
        return self.message
    }

    func setPeerID(_ peerID: String?) {
        self.peerID = peerID
    }

    func getPeerID() -> String? {
        return self.peerID
    }
}
